app/components/CoinListCell/CoinListCell.js

```diff
@@ -12,73 +16,100 @@ export default class CoinListCell extends Component {
-                </Text>
-              </View>
-            </View>
+    return (
+      <View style={s.mainContainer}>
+        <TouchableOpacity onPress={this.handleCellPress}>
+          <View style={s.contentContainer}>
+            <View style={s.collapsedContainer}>
+              <View style={s.nameSymbolContainer}>
+                <Image
+                  style={s.symbolImage}
+                  source={{ uri: `https://chasing-coins.com/api/v1/std/logo/${symbol}` }}
+                />
 
-            <View style={coinListCellStyles.priceContainer}>
-              <Text style={coinListCellStyles.priceText}>
-                    ${this.props.item.price_usd}
-              </Text>
-              <View style={isPercentChangeNegative ? [coinListCellStyles.changeContainer,coinListCellStyles.changeContainerNegative]:
-                coinListCellStyles.changeContainer
-                }
-              >
-                <Text style={
-                    isPercentChangeNegative ?
-                    [coinListCellStyles.changeText, coinListCellStyles.changeTextNegative]:
-                    coinListCellStyles.changeText}
-                >
-                  {this.props.item.percent_change_24h}%
-                </Text>
+                <View style={s.nameContainer}>
+                  <Text style={s.nameText}>
+                    {name}
+                  </Text>
+                  <Text style={s.symbolText}>
+                    {symbol}
+                  </Text>
+                </View>
               </View>
+              {
+                this.state.isExpanded === false ?
+                  <View style={s.priceContainer}>
+                    <Text style={isNegative ?
+                    [s.priceText,
+                    s.priceTextNegative] :
+                    s.priceText}
+                    >
+                      ${priceUSD}
+                    </Text>
+                    <CoinListCellChangeButton
+                      isNegative={isNegative}
+                      percentChange={percentChange24h}
+                      priceChange={priceChange}
+                    />
+                  </View> :
+                  <CoinListCellFavoritesButton />
+              }
             </View>
+            { this.state.isExpanded &&
+              <View style={s.dividerLine} />
+            }
+            { this.state.isExpanded &&
+              <CoinListCellRangeView item={this.props.item} />
+            }
+            {
+              this.state.isExpanded &&
+              <CoinListCellChangeView item={this.props.item} />
+            }
+            { this.state.isExpanded &&
+              <CoinListCellStatsView item={this.props.item} />
+            }
           </View>
-          {expandedView}
-        </View>
-      </TouchableOpacity>
-    </View>
-    );
+        </TouchableOpacity>
+      </View>
+    )
   }
 }
+
+CoinListCell.defaultProps = {
+  item: {
+    percent_change_24h: '-',
```

Comment:
This is where the mapping of the data might come in.
Now we're using `percent_change_24h` in a couple of places. If the API changes it's keys, we brake EVERYWHERE. When we do the mapping, all of the internal mappings will be the same and if there is a change we will only need to replace it in 1 place.
When we'll do this, we'll need to see if it impacts performance. Remapping 100 objects shouldn't be difficult and it can be a part of the `Loading` time.
I think we'll be fine and I can take care of that next. Just a thought that came up.

Reply:
👍